jujugui/static/gui/src/app/jujulib/register.js

```diff
@@ -171,6 +173,123 @@ var module = module;
+            - name {String} The user provided identifier of the card
+            - addressLine1 {String|Null} The first address line
+            - addresssLine2 {String|Null} The second address line
+            - addressCity {String|Null} The address city
+            - addressState {String|Null} The address state
+            - addressZip {String|Null} The address post code
+            - addressCountry {String|Null} The address country
+            - country {String} The cardholder's country
+            - expMonth {String} The two digit number of the month the card
+              expires
+            - expYear {String} The two or four digit number of the year the card
+            - last4 {String} The last four digits of the card number
+            - object {String} The type of object e.g. "card"
+            - funding {String} The card credit type e.g. "credit"
+          - created {String} The timestamp for when the token was created.
+          - livemode {Boolean} Whether this token was created using a live API
+            key.
+          - type {String} The token type e.g. "card".
+          - object {String} The type of object, in this case always "token".
+          - used {Boolean} Whether this token has been used.
+    */
+    createToken: function(card, callback) {
+      const handler = (status, response) => {
+        if (response && response.error) {
+          callback(response.error.message, null);
+          return;
+        }
+        const card = response.card;
+        const token = {
+          id: response.id,
+          card: {
+            name: card.name || null,
+            addressLine1: card['address_line1'] || null,
+            addressLine2: card['address_line2'] || null,
+            addressCity: card['address_city'] || null,
+            addressState: card['address_state'] || null,
+            addressZip: card['address_zip'] || null,
+            addressCountry: card['address_country'] || null,
+            country: card.country,
+            expMonth: card['exp_month'],
+            expYear: card['exp_year'],
+            last4: card.last4,
+            object: card.object,
+            brand: card.brand,
+            funding: card.funding
+          },
+          created: response.created,
+          livemode: response.livemode,
+          type: response.type,
+          object: response.object,
+          used: response.used
+        };
+        callback(null, token);
+      };
+      const data = {
+        number: card.number,
+        cvc: card.cvc,
+        exp_month: card.expMonth,
+        exp_year: card.expYear,
+        name: card.name,
+        address_line1: card.addressLine1,
+        address_line2: card.addressLine2,
+        address_city: card.addressCity,
+        address_state: card.addressState,
+        address_zip: card.addressZip,
+        address_country: card.addressCountry
+      };
+      this.stripe.card.createToken(data, handler);
```

Comment:
this assumes a third party library loaded successfully, which we can't be sure about as the GUI can be run in a network restricted environment. We'll need to know about this so that we can update the UI accordingly elsewhere so maybe you should add another method which is called at the start of every method requiring stripe.

```Javascript
stripeInitialized: function(callback) {
  if (!this.stripe) {
    callback('Stripe does not exist, cannot perform request.');
    return false;
  }
  return true;
}
``` 

And then this would change to

```JavaScript
if (this.stripeInitialized(callback)) {
  this.stripe.card.createToken(data, handler);
}
```